vars/buildDockerAndPublishImage.groovy

```diff
@@ -3,29 +3,69 @@ import java.text.SimpleDateFormat
 import java.util.Date
 import java.text.DateFormat
 
+def makecall(String action, String imageDeployName, String targetOperationSystem, String SpecificDockerBakeFile) {
+  final String jenkinsInfraBakeFile = 'jenkinsinfrabakefile.hcl'
+  final String bakefileContent = libraryResource 'io/jenkins/infra/docker/' + jenkinsInfraBakeFile
+  if (action != 'build' && action != 'deploy'){
+    echo 'ERROR: makecall need an action parameter with value: `build` or `deploy`.'
+    currentBuild.result = 'FAILURE'
+    return
+  }
+  if (imageDeployName == '') {
+    echo 'ERROR: makecall need an action imageDeployName.'
+    currentBuild.result = 'FAILURE'
+    return
+  }
+  if (targetOperationSystem == '') {
+    echo 'ERROR: targetOperationSystem cannot be empty.'
+    currentBuild.result = 'FAILURE'
+    return
+  }
+  // Please note that "make deploy" and the generated bake deploy file uses the environment variable "IMAGE_DEPLOY_NAME"
+  withEnv(["IMAGE_DEPLOY_NAME=${imageDeployName}"]) {
+    if (isUnix()) {
+      if (SpecificDockerBakeFile) {
+        withEnv(["DOCKER_BAKE_FILE=${SpecificDockerBakeFile}"]) {
+          sh 'make bake-$action'
```

Comment:
```suggestion
        withEnv(["DOCKER_BAKE_FILE=${SpecificDockerBakeFile}", "ACTION=${action}"]) {
          sh 'make "bake-${ACTION}"'
```

=> you have to pass the value of `$action` groovy variable to the shell environment (or use double quotes so that groovy interpolates its `$action` variable directly passing its value to the `sh` step).

=> Don't forget to update the unit test to check for the value passed to `withEnv` instead of `sh` if you take my suggestion

Reply:
I choose the groovy interpolation